main/src/cgeo/geocaching/storage/DataStore.java

```diff
@@ -1477,11 +1477,11 @@ public static boolean deleteWaypoint(final int id) {
     }
 
     private static void saveSpoilersWithoutTransaction(final Geocache cache) {
-        final String geocode = cache.getGeocode();
-        database.delete(dbTableSpoilers, "geocode = ?", new String[]{geocode});
-
         final List<Image> spoilers = cache.getSpoilers();
         if (CollectionUtils.isNotEmpty(spoilers)) {
+            final String geocode = cache.getGeocode();
+            database.delete(dbTableSpoilers, "geocode = ?", new String[]{geocode});
```

Comment:
Just wondering if the deletion performance would also benefit from a preparedStatement. (The existing prepared statements were mostly introduced to avoid that the database query cache has an overflow, not because of performance reasons).